src/app/modules/core/core.errors.ts

```diff
@@ -73,3 +73,12 @@ export class MissingFeatureError extends ApplicationError {
     )
   }
 }
+
+/**
+ * Error thrown when attachment upload fails
+ */
+export class AttachmentUploadError extends ApplicationError {
+  constructor(message?: string) {
```

Comment:
could you provide a simple default error message here (e.g. `constructor(message = 'Error while uploading encrypted attachments to S3')`? this is to make our logging clearer